src/core/execute.c

```diff
@@ -2089,9 +2089,7 @@ static int setup_exec_directory(
                         }
                 } else {
                         r = mkdir_label(p, context->directories[type].mode);
-                        if (r == -EEXIST)
-                                continue;
-                        if (r < 0)
+                        if (r < 0 && r != -EEXIST)
```

Comment:
this undoes a central part of 30c81ce2cef97b05e143c7adf4cd1b1c5fb59932.

@keszybz can you comment?

Reply:
Oh, I forgot the commit. Hmm. So, how should we migrate runtime directories? Should we add a flag to `RuntimeDirectory=`?

Reply:
What do you think about `chown`ing runtime directories only when `DynamicUser=` is set?

Reply:
I think this is OK. In 30c81ce2cef97b05e143c7adf4cd1b1c5fb59932 I didn't think about this case (runtime directory of a dynamic user when this user gets a new number).